newsroom/news_api/news/search_service.py

```diff
@@ -1,669 +1,123 @@
-        "version",
-        "where",  # ? Should we support this in original API?
-        "q",  # ** in original
-        "default_operator",  # ** in original
-        "filter",  # ** in original
-        "service",
-        "subject",
-        "genre",
-        "urgency",
-        "priority",
-        "type",
-        "item_source",
-        "sort",  # ** in original
-        "timezone",  # alias=timezone_offset?? Double check this one
-        "products",  # alias=product (this version support list of products ID, might also support that in original)
-    }
-
-    default_sort = [{"versioncreated": "desc"}]
-
-    # set of fields that are allowed to be excluded in the exlude_fields parameter
-    allowed_exclude_fields = {
-        "version",
-        "versioncreated",
-        "firstcreated",
-        "headline",
-        "byline",
-        "slugline",
-    }
-
-    # set of fields that can be specified in the include_fields parameter
-    allowed_include_fields = {
-        "type",
-        "urgency",
-        "priority",
-        "language",
-        "description_html",
-        "located",
-        "keywords",
-        "source",
-        "subject",
-        "place",
-        "wordcount",
-        "charcount",
-        "body_html",
-        "readtime",
-        "profile",
-        "service",
-        "genre",
-        "associations",
-    }
-
-    default_fields = {
-        "_id",
-        "uri",
-        "embargoed",
-        "pubstatus",
-        "ednote",
-        "signal",
-        "copyrightnotice",
-        "copyrightholder",
-        "versioncreated",
-        "evolvedfrom",
-        "original_id",
-    }
-
-    # set of fields that will be removed from all responses, we are not currently supporting associations and
-    # the products embedded in the items are the superdesk products
-    mandatory_exclude_fields = {"_current_version", "products"}
+from .types import NewsApiSearchRequestArgs
 
-    section = "news_api"
-    limit_days_setting = "news_api_time_limit_days"
 
-    def get(self, req, lookup):
-        resp = super().get(req, lookup)
+default_search_filters = [
+    prefill_company,
+    prefill_products,
+    apply_section_filter,
+    apply_company_filter,
+    apply_products_filter,
+    apply_filter_fields,
+    apply_date_filter,
+    apply_request_filter,
+    apply_projection,
+    validate_page,
+]
 
-        orig_request_params = getattr(req, "args", MultiDict())
 
-        # Can't get the exclude projection to work do pop the exclude fields here
-        exclude_fields = (
-            self.mandatory_exclude_fields.union(set(orig_request_params.get("exclude_fields").split(",")))
-            if orig_request_params.get("exclude_fields")
-            else self.mandatory_exclude_fields
-        )
+class NewsApiSearchServiceAsync(BaseNewshubSearchService[NewsApiSearchRequestArgs, WireItem]):
+    search_args_class = NewsApiSearchRequestArgs
+    filters = cast(list[SearchFilterFunction], default_search_filters)
```

Comment:
If you defined the type of `default_search_filters` above, then this `cast` here would not be required (and allows mypy to pick up type errors for these as well)